src/extensions/mountainview/MountainView/ViewContainer.tsx

```diff
@@ -1,46 +1,117 @@
+                }
+                else if (v.area === 'south') {
+                    setCurrentSouthView(v)
+                    setActiveArea('south')
+                }
             }
         })
-    }, [views])
+    }, [views, activeArea])
+
+    const H0 = (height - tabBarHeight * 2) / 2
+    const areas: {[key: string]: {
+        tabBarStyle: React.CSSProperties,
+        divStyle: React.CSSProperties
+    }} = {
+        'north': {
+            tabBarStyle: { left: 0, top: 0, width: width, height: tabBarHeight },
+            divStyle: {left: 0, top: tabBarHeight, width: width, height: H0}
+        },
+        'south': {
+            tabBarStyle: { left: 0, top: tabBarHeight + H0, width: width, height: tabBarHeight },
+            divStyle: { left: 0, top: tabBarHeight * 2 + H0, width: width, height: H0 }
+        }
+    }
+
+    const handleClick = useCallback((e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+        const y = eventToPos(e)[1]
+        const newActiveArea = (y < tabBarHeight + H0) ? 'north' : 'south'
+        if (newActiveArea !== activeArea) {
+            setActiveArea(newActiveArea)
+        }
+    }, [H0, activeArea, setActiveArea])
+
+    const handleDragOver = useCallback((event: React.DragEvent<HTMLDivElement>) => {
+        // the following is needed otherwise we can't get the drop. See: https://stackoverflow.com/questions/50230048/react-ondrop-is-not-firing
+        event.stopPropagation();
+        event.preventDefault();
+    }, [])
+
+    const handleDragDrop = useCallback((e: React.DragEvent<HTMLDivElement>) => {
+        const y = eventToPos(e)[1]
+        const dropArea = (y < tabBarHeight + H0) ? 'north' : 'south'
+        const viewId = e.dataTransfer.getData('viewId')
+        if (viewId) {
+            const view = views.filter(v => v.viewId === viewId)[0]
+            if (view) {
+                if (view.area !== dropArea) {
+                    onSetViewArea(view, dropArea)
+                }
+            }
+        }
+    }, [views, H0, onSetViewArea])
+
+
     if (!Array.isArray(children)) {
-        throw Error('Unepected children in ViewContainer')
+        throw Error('Unexpected children in ViewContainer')
     }
     const children2 = children as React.ReactElement[]
-    const W = width || size.width || 300
+    const W = width || 300
     return (
-        <div style={{width: W}}>
-            <ViewContainerTabBar
-                views={views}
-                currentView={currentView}
-                onCurrentViewChanged={setCurrentView}
-                onViewClosed={onViewClosed}
-            />
+        <div
+            style={{width: W, height: height}} onClick={handleClick}
+            onDragOver={handleDragOver}
+            onDrop={handleDragDrop}
+        >
+            <div key="north-tab-bar" style={{position: 'absolute', ...areas['north'].tabBarStyle, border: 1, borderStyle: 'solid', borderColor: (activeArea === 'north') ? 'black' : 'white'}}>
```

Comment:
A div element for the north tab bar and a div element for the south bar, positioned absolutely.